src/test/java/core/basesyntax/RegistrationValidationTest.java

```diff
@@ -0,0 +1,148 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.exception.RegistrationException;
+import core.basesyntax.model.User;
+import core.basesyntax.service.RegistrationService;
+import core.basesyntax.service.RegistrationServiceImpl;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class RegistrationValidationTest {
+    private static final StorageDao storageDao = new StorageDaoImpl();
+    private RegistrationService registrationService;
+
+    //region Repeated Tasks
+    @BeforeEach
+    void setUp() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @AfterEach
+    void tearDown() {
+        Storage.people.clear();
+    }
+    //endregion
+
+    //region assertEquals Tests
+    @Test
+    void register_validUser_Ok() {
+        User validUser = new User();
+        validUser.setAge(18);
+        validUser.setLogin("validLogin");
+        validUser.setPassword("validPassword");
+        User actual = registrationService.register(validUser);
+        assertEquals(actual, validUser);
+    }
+
+    @Test
+    void getUser_AfterAddFromStorageClass_Ok() {
+        User user = new User();
+        user.setAge(19);
+        user.setLogin("test007");
+        user.setPassword("qwerty1234");
+        Storage.people.add(user);
+        User getUser = storageDao.get(user.getLogin());
+        assertEquals(user, getUser);
+    }
+    //endregion
+
+    //region assertThrows Tests
+    @Test
+    void register_invalidUser_notOk() {
+        User invalidAgeUser = new User();
+        invalidAgeUser.setAge(17);
+        invalidAgeUser.setLogin("validName");
+        invalidAgeUser.setPassword("validPassword");
+        assertThrows(RegistrationException.class,
+                () -> registrationService.register(invalidAgeUser));
+
+        User invalidLoginUser = new User();
+        invalidLoginUser.setAge(18);
+        invalidLoginUser.setLogin("");
+        invalidLoginUser.setPassword("validPassword");
+        assertThrows(RegistrationException.class,
+                () -> registrationService.register(invalidLoginUser));
+
+        User invalidPasswordUser = new User();
+        invalidPasswordUser.setAge(19);
+        invalidPasswordUser.setLogin("validName");
+        invalidPasswordUser.setPassword("abc");
+        assertThrows(RegistrationException.class,
+                () -> registrationService.register(invalidPasswordUser));

```

Comment:
The test 'register_invalidUser_notOk' should be split into separate tests for each invalid case to ensure test independence and clear identification of which case fails. Each test should have a single responsibility.